tests/Routing/UrlGeneratorTest.php

```diff
@@ -0,0 +1,117 @@
+<?php
+
+/*
+ * This file is part of Contao.
+ *
+ * Copyright (c) 2005-2016 Leo Feyer
+ *
+ * @license LGPL-3.0+
+ */
+
+namespace Contao\CoreBundle\Test\Routing;
+
+use Contao\CoreBundle\Routing\UrlGenerator;
+
+/**
+ * Tests the UrlGenerator class.
+ *
+ * @author Andreas Schempp <https://github.com/aschempp>
+ */
+class UrlGeneratorTest extends \PHPUnit_Framework_TestCase
+{
+
+    /**
+     * Tests the object instantiation.
+     */
+    public function testInstantiation()
+    {
+        $this->assertInstanceOf('Contao\CoreBundle\Routing\UrlGenerator', $this->generator());
+    }
+
+    public function testRoute()
+    {
+        $this->assertEquals('contao_index', $this->generator(false, 0)->generate('index'));
+        $this->assertEquals('contao_index', $this->generator(true, 0)->generate('index'));
+        $this->assertEquals('contao_frontend', $this->generator(false, 0)->generate('foobar'));
+        $this->assertEquals('contao_frontend', $this->generator(true, 0)->generate('foobar'));
+        $this->assertEquals('contao_frontend', $this->generator(false, 0)->generate('foobar/test'));
+    }
+
+    public function testWithoutParameters()
+    {
+        $this->assertEquals('foobar', $this->generator()->generate('foobar')['alias']);
+        $this->assertEquals('foobar/test', $this->generator()->generate('foobar/test')['alias']);
+        $this->assertEquals('foobar/article/test', $this->generator()->generate('foobar/article/test')['alias']);
```

Comment:
Its supported in PHP 5.4 ;-)
http://php.net/manual/en/language.types.array.php#example-102